arangod/Aql/RemoteExecutor.h

```diff
@@ -132,6 +133,10 @@ class ExecutionBlockImpl<RemoteExecutor> : public ExecutionBlock {
   /// @brief the last remote response Result object, may contain an error.
   arangodb::Result _lastError;
 
+  /// @brief Mutex to cover against the race, that a getSome request
+  ///        is responded before the ticket id is registered.
+  arangodb::Mutex _communicationMutex;
+
   OperationID _lastTicketId;
```

Comment:
Maybe it would be advantageous to make more clear which members are prone to races, at least with additional comments to both the mutex and all affected members.

Reply:
i.e. `_lastError`, `_lastResponse`, `_lastTicketId`

Reply:
Shutdown is correct. Fixed now.

The logic is as follows:
ticketID == 0 indicates no request should be handled.
And ticketIDs are unique.

Now the Mutex protects the ticketId, and guarantees that we only handle the correct request, and ignore incorrect ones.
WIthout the mutex the ticketid could be incorrect in the race situations.